diff --git a/packages/cli/src/controllers/ai.controller.ts b/packages/cli/src/controllers/ai.controller.ts
index 24510311..f4516096 100644
--- a/packages/cli/src/controllers/ai.controller.ts
+++ b/packages/cli/src/controllers/ai.controller.ts
@@ -303,6 +303,13 @@ export class AiController {
 		_: Response,
 	): Promise<AiAssistantSDK.BuilderInstanceCreditsResponse> {
 		try {
+			if (!this.workflowBuilderService.isProxyConfigured()) {
+				return {
+					creditsQuota: -1,
+					creditsClaimed: 0,
+				};
+			}
+
 			return await this.workflowBuilderService.getBuilderInstanceCredits(req.user);
 		} catch (e) {
 			assert(e instanceof Error);
